### PR TITLE
feat(popupfield): Add header and aria label to dialog and refactor fieldButton component. Add more descriptive documentation.

### PR DESCRIPTION
**Pull request by rachmulvey**
…ent. Add more descriptive documentation.

---
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/34)

### CODE REVIEW

File: src/components/PopupField/index.js [hunk 1]
```diff
@@ -122,28 +128,29 @@ class PopupField extends Component {
...
+              <Header
+                setFocusElementRef={this.setFocusElementRef}
+                closePopup={this.closePopup}
+                headerLabel={headerLabel}
+                headerHeight={headerHeight}
+                HeaderIcon={HeaderIcon}
+                closeAriaLabel={closeAriaLabel}
+                renderHeaderChildren={renderHeaderChildren}
+              />
+              <div css={{ overflow: 'auto' }}>{content}</div>
```
Comment (Pallandor) [inline src/components/PopupField/index.js]: If we're enforcing `layout.gutter` spacing dimensions in the `Header` component, we should probably apply the same to the containing div node of content, so the consumer doesn't have to manually fix alignments by hand.

Comment (rachmulvey) [reply src/components/PopupField/index.js]: thanks, made some updates

File: src/components/PopupField/__tests__/Button.spec.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  let component;
+
+  it('renders correctly with defaults', () => {
+    component = mount(<Button onClick={() => {}} setButtonRef={() => {}} />);
+
+    expect(component).toMatchSnapshot();
+  });
+
+  it('renders correctly with props provided', () => {
+    component = mount(
```
Comment (yogesh-jankin) [inline src/components/PopupField/__tests__/Button.spec.js]: please, can we use shallow instead of mount?

Comment (rachmulvey) [reply src/components/PopupField/__tests__/Button.spec.js]: updated

**Pull request merged**